pyaedt/emit.py

```diff
@@ -1,16 +1,84 @@
 from pyaedt.application.AnalysisEmit import FieldAnalysisEmit
+from pyaedt.generic.general_methods import pyaedt_function_handler
+from pyaedt import generate_unique_project_name
+from pyaedt import settings
+from pyaedt.generic.general_methods import pyaedt_function_handler
+import sys
+import os
+supported_versions = [2023.1, 2023.2]
+sys.path.append("C:\\Users\\cchandel\\workspace\\build_output\\64Release\\Delcross")
+import EmitApiPython
+class Revision:
+    '''
+    Provides the Revision object.
+
+    Parameters
+    ----------
+    Emit_obj :
+        This is the Emit Object to which this revision is associated with.
+    name : str, optional
+        Name of the Revision which is about to be created. The default is ``None``,
+        in which case - a default name is given.
+    Examples
+    --------
+    Create an instance of Emit, you can choose to define any of the parameters of Emit here.
+    >>> aedtapp = Emit()
+    >>> rev = Revision(aedtapp, "Revision 1")
+    >>> eng = aedtapp.get_engine()
+    >>> domain = EmitApiPython.InteractionDomain()
+    >>> rev.run(domain, eng)
+    '''
+    @pyaedt_function_handler()
+    def __init__(self, Emit_obj, name = ""):
+        subfolder =''
+        for f in os.scandir(Emit_obj.location):
+            if os.path.splitext(f.name)[1].lower() in '.aedtresults':
+                subfolder = os.path.join(f.path, "EmitDesign1")
+        default_behaviour = not os.path.exists(os.path.join(subfolder, "{}.emit".format(name)))
+        if default_behaviour:
+            print("The most recent revision that was generated will be used because the revision you specified does not exist.")
+        if(name == "" or default_behaviour):
+            file = max([f for f in os.scandir(subfolder)], key=lambda x: x.stat().st_mtime)
+            full = file.path
+            name = file.name
+        else:
+            full = subfolder + '/{}.emit'.format(name)
+        self.name = name
+        self.path = full
+        self.emit_obj = Emit_obj
+    @pyaedt_function_handler()
+    def run(self, domain, eng):
+        """
+        Load the revision, and proceed to analyze along the given domain
+
+        Parameters
+        ----------
+            domain:
+                An InteractionDomain must be defined to constrain the analysis parameters.
+            eng: 
+                An EmitEngine object from the associated Emit object must be given (temporary solution).
+        Returns
+        -------
+        :class:`Interaction`
+            Interaction object.
 
+        References
+        ----------
+        >>> eng = aedtapp.get_engine()
+        >>> domain = EmitApiPython.InteractionDomain()
+        >>> rev.run(domain, eng)
 
+        """
+        self.emit_obj.load_revision(self.path)
+        print("checkpoint - results loaded")
+        t = eng.analyze(domain)
+        print("checkpoint - interaction generated")
+        return t
 class Emit(FieldAnalysisEmit, object):
```

Comment:
I think a blank line is needed between lines 78 and 79.

Reply:
Resolved!